server/schedulers/hot_region.go

```diff
@@ -476,40 +571,51 @@ func (h *balanceHotRegionsScheduler) balanceByLeader(cluster opt.Cluster, stores
 			continue
 		}
 
+		if _, ok := h.pendingOpInfos[srcRegion.GetID()]; ok {
+			continue
+		}
+
 		if !opt.IsHealthyAllowPending(cluster, srcRegion) {
 			schedulerCounter.WithLabelValues(h.GetName(), "unhealthy-replica").Inc()
 			continue
 		}
 
 		filters := []filter.Filter{filter.StoreStateFilter{ActionScope: h.GetName(), TransferLeader: true}}
-		candidateStoreIDs := make([]uint64, 0, len(srcRegion.GetPeers())-1)
+		filters = append(filters, filter.NewHealthFilter(h.GetName()))
+		candidateStoreIDs := make(map[uint64]struct{}, len(srcRegion.GetPeers())-1)
 		for _, store := range cluster.GetFollowerStores(srcRegion) {
 			if !filter.Target(cluster, store, filters) {
-				candidateStoreIDs = append(candidateStoreIDs, store.GetID())
+				candidateStoreIDs[store.GetID()] = struct{}{}
 			}
 		}
 		if len(candidateStoreIDs) == 0 {
 			continue
 		}
-		destStoreID := h.selectDestStore(candidateStoreIDs, rs.GetBytesRate(), srcStoreID, storesStat)
-		if destStoreID == 0 {
-			continue
-		}
 
-		destPeer := srcRegion.GetStoreVoter(destStoreID)
-		if destPeer != nil {
-			h.leaderLimit = h.adjustBalanceLimit(srcStoreID, storesStat)
+		regionBytesRate := rs.GetBytesRate()
+		if balanceType == hotWrite {
+			dstStoreID = h.selectDstStoreByHot(candidateStoreIDs, regionBytesRate, srcStoreID, storesStat)
+		} else {
+			dstStoreID = h.selectDstStoreByScore(candidateStoreIDs, regionBytesRate, balanceType)
+		}
 
-			return srcRegion, destPeer, Influence{ByteRate: rs.GetBytesRate()}
+		if dstStoreID != 0 {
+			dstPeer := srcRegion.GetStoreVoter(dstStoreID)
+			if dstPeer != nil {
+				h.leaderLimit = h.adjustBalanceLimit(srcStoreID, storesStat)
+				return srcRegion, dstPeer, Influence{ByteRate: rs.GetBytesRate()}
+			}
 		}
 	}
 	return nil, nil, Influence{}
 }
 
+const pickByFlowBytes = false
```

Comment:
what's the purpose? always false?

Reply:
It is used to determine the priority when sorting. I am not sure if we need to add it to the config. cc @Luffbee

Reply:
I think we can remove it and corresponding code.

Reply:
Well